tests/ui/resolve/open-ns-1.rs

```diff
@@ -0,0 +1,19 @@
+//@ build-pass
+//@ aux-crate:my_api=open-ns-my_api.rs
+//@ aux-crate:my_api::utils=open-ns-my_api_utils.rs
+//@ aux-crate:my_api::core=open-ns-my_api_core.rs
+//@ compile-flags: -Z namespaced-crates
+//@ edition: 2024
+
+//@ ignore-test FIXME(packages_as_namespaces): feature not implemented yet
+
+use my_api::root_function;
+use my_api::utils::util;
```

Comment:
This should be an ambiguity error.
`my_api::utils` refers both to `my_api(crate)::utils(mod)` and `my_api(open_ns)::utils(crate)`, and they have different definitions.

Alternatively `my_api(crate)::utils(mod)` will shadow `my_api(open_ns)::utils(crate)` in the 3-scope model, if we do not report ambiguities even for explicit uses.

Reply:
Ah, okay, I mixed up `tests/ui/resolve/auxiliary/open-ns-mod-my_api.rs` and `tests/ui/resolve/auxiliary/open-ns-mod-my_api.rs`, the files are named too similarly.
I'll need to re-review all the cases.